ops/jenkins/bfd-build-platinum/Jenkinsfile

```
@@ -9,6 +9,70 @@
 // These variables are accessible throughout this file (except inside methods and classes).
 def goldAmiId
 def platinumAmiId
+def currentStage
+
+def buildPlatinum(String platinumId = '', String goldId = '') {
+    // Retrieve latest platium and assign the sourceAmiId tag value with the latest gold AMI ID.
+    withCredentials([file(credentialsId: 'bfd-vault-password', variable: 'vaultPasswordFile')]) {
+        // packer is always run from $repoRoot/ops/ansible/playbooks-ccs
+        dir('ops/ansible/playbooks-ccs') {
+            env.goldAmi = goldId
+            echo "Using the folllowing gold image: ${env.goldAmi}"
+
+            sh '''
+            packer build -color=false -var vault_password_file="$vaultPasswordFile" \
+            -var source_ami="${goldAmi}" \
+            -var subnet_id=subnet-092c2a68bd18b34d1 \
+            ../../packer/build_bfd-platinum.json
+            '''
```

It might be simpler to avoid setting environment variables from input. What do you think of this?
```suggestion
            echo "Using the following gold image: ${goldId}"

            sh """
            packer build -color=false -var vault_password_file="$vaultPasswordFile" \
            -var source_ami=${goldId} \
            -var subnet_id=subnet-092c2a68bd18b34d1 \
            ../../packer/build_bfd-platinum.json
            """
```